docs: add a paragraph on fingerprint generation (#12464)
hashicorp/packer: website/content/docs/hcp/index.mdx

Packer uses a unique fingerprint for tracking the completion of builds associated to an iteration. By default a fingerprint is automatically generated by Packer during each invocation of `packer build`, unless a fingerprint is manually provided via the `HCP_PACKER_BUILD_FINGERPRINT` environment variable.

> In versions before 1.9.0, this fingerprint was computed from the git SHA of the current HEAD in which your template is stored. If you are running builds outside of Git, you would have to set the `HCP_PACKER_BUILD_FINGERPRINT` environment variable prior to invoking `packer build`.

Reviewer: ```suggestion
In versions before 1.9.0, this fingerprint was computed from the Git SHA of the current HEAD in which your template is stored. If you are running builds outside of Git, you had to set the `HCP_PACKER_BUILD_FINGERPRINT` environment variable prior to invoking `packer build`.
```

> Starting with Packer 1.9.0, the fingerprint generation does not rely on Git at all, and instead Packer will generate a Unique Lexicographically sortable IDentifier (ULID).

Reviewer: ```suggestion
Starting with Packer 1.9.0, fingerprint generation does not rely on Git at all, and instead Packer now generates a Unique Lexicographically sortable Identifier (ULID) as the fingerprint for every `packer build` invocation.
```
